sources/packages/api/src/esdc-integration/disbursement-receipt-integration/disbursement-receipt-request.service.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable } from "@nestjs/common";
+import { InjectLogger } from "../../common";
+import { LoggerService } from "../../logger/logger.service";
+import {
+  ConfigService,
+  DisbursementReceiptService,
+  ReportService,
+} from "../../services";
+import { ESDCFileHandler } from "../esdc-file-handler";
+import { DailyDisbursementUploadResult } from "./models/disbursement-receipt-integration.model";
+import { ReportsFilterModel } from "src/services/report/report.models";
+import { DisbursementReceiptIntegrationService } from "./disbursement-receipt-integration.service";
+
+@Injectable()
+export class DisbursementReceiptRequestService extends ESDCFileHandler {
+  constructor(
+    configService: ConfigService,
+    private readonly reportService: ReportService,
+    private readonly disbursementReceiptService: DisbursementReceiptService,
+    private readonly integrationService: DisbursementReceiptIntegrationService,
+  ) {
+    super(configService);
+  }
+
+  /**
+   * 1. Fetches the Daily disbursement information which are not sent.
+   * 2. Create a csv file of the total records sent on the process date.
+   * 3. Upload the content to the zoneB SFTP server.
+   * @param batchRunDate Date in which the daily disbursement records needs to be fetched.
+   * @returns Processing Daily Disbursement request result.
+   */
+  async processProvincialDailyDisbursements(
```

Comment:
Are we not missing some error handling along this method to provide a better log output in case something goes wrong?